src/components/typography/typography.tsx

```diff
@@ -0,0 +1,36 @@
+import { useTheme } from '@react-navigation/native';
+import React from 'react';
+import { Text, TextProps } from 'react-native';
+
+import { styles } from './typography.styles';
+
+interface TypographyProps extends TextProps {
+  variant?: 'h1' | 'h2' | 'h3' | 'body' | 'caption';
+  style?: object;
+}
+
+/**
```

Comment:
Can you cross check the content of your comments here and other places, based on guidance here - [How to write better code comments](https://dev.to/adammc331/todo-write-a-better-comment-4c8c)?